gradesa/src/components/ui/sidenavigation.js

```diff
@@ -0,0 +1,40 @@
+'use client'
+
+import { useState } from 'react';
+import Link from 'next/link'
+import "./sidebar.css";
+
+const Sidebar = () => {
+    const [isDropdownOpen, setDropdownOpen] = useState(false);
+
+    const toggleDropdown = () => {
+        setDropdownOpen(!isDropdownOpen);
```

Comment:
```suggestion
        setDropdownOpen(previousState => !previousState)
```
State updates are asynchronous, if you want to do something based on a previous state it's better to use this callback pattern.

100% won't cause an issue here, but it's a good habit.